plugins/plugin-hosted/codenvy-machine-hosted/src/main/java/com/codenvy/machine/UriTemplateServerProxyTransformer.java

```diff
@@ -64,21 +70,46 @@ public ServerImpl transform(ServerImpl server) {
         if (serverPath.startsWith("/")) {
             serverPath = serverPath.substring(1);
         }
-
+        final String externalAddress;
+        final String externalServerHost;
+        final String internalServerHost;
+        if (!Strings.isNullOrEmpty(externalAddressProperty) && !Strings.isNullOrEmpty(codenvyHost)) {
```

Comment:
As far as I can see codenvyHost is set from puppet variable codenvy::host_url. @riuvshin is it always set?
If so then this variable is always non-null.